torchgeo/datasets/bigearthnetv2.py

```diff
@@ -0,0 +1,664 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""BigEarthNetv2 dataset."""
+
+import glob
+import json
+import os
+from typing import Callable, Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+import rasterio
+import torch
+from matplotlib.figure import Figure
+from rasterio.enums import Resampling
+from torch import Tensor
+import pandas as pd
+from torchvision import transforms
+from torchgeo.datasets.geo import NonGeoDataset
+from torchgeo.datasets.utils import download_url, extract_archive, sort_sentinel2_bands
+
+class BigEarthNetv2(NonGeoDataset):
```

Comment:
I think between the existing `BigEarthNet` version and version 2, there is a lot of shared code. The `__init__` signature is the same as well, so I would think it's beneficial to inherit from `BigEarthNet` instead of `NonGeoDataset` so that a lot of the code does not have to be written twice, and instead only overwrite the relevant parts.

Reply:
Can someone point me to a summary of the changes between v1 and v2? Basically I'm curious if this is just a newer version of the same dataset (slightly more images or better labels) or a completely new iteration. We could either do:
```python
BigEarthNet(version=1)
BigEarthNet(version=2)
```
in the case of the former or:
```python
BigEarthNet()
BigEarthNetv2()
```
in the case of the latter.

Reply:
Yes you're right, I've basically used v1 as the template to write v2. 
v2 is the same dataset with the following changes:

- Better labels
- Better spatial splits
- Fewer images - problematic images removed

The [reBEN paper](http://arxiv.org/abs/2407.03653) details these differences

Reply:
Gotcha. In that case I'm kind of leaning towards adding a `version` parameter to the existing dataset. We could default to 2. This would require the fewest code modifications.

Reply:
We have a similar discussion on naming in #2394